### DISCUSSION TITLE
How do I set parameters for the `raw` text in a template-config?

### DISCUSSION THREAD

**Discussion by exo-cortex** [typst/typst#3252] [Q&A]
I know how to set parameters in my typst-doc itself (by using "show"), but I don't understand how to set it i.e. the fill-value of the block of raw-text inside of my separate config-file.

**Comment by Julian-Wassmann** [comment]
You can follow the tutorial for [Making a Template](https://typst.app/docs/tutorial/making-a-template/) and adapt it for your needs. For your example with `raw` that would be:

**main.typ**
````typst
#import "template.typ": template

#show: template

```typ
= Chapter 1
#let hi = "Hello World"
```
````
**template.typ**
```typst
#let template(doc) = {
  // example from https://typst.app/docs/reference/text/raw/#parameters-theme
  show raw: it => block(
    fill: rgb("#1d2433"),
    inset: 8pt,
    radius: 5pt,
    text(fill: rgb("#a2aabc"), it)
  )
  
  doc
}
```
Reproduction URL: https://typst.app/project/rfCMWbobw1L5tAcLCLYq-m

Is this what you are looking for?
